Detaljside for observasjon

jorgkv opened the pull request:
Endringen skal legge til en detaljside for observasjoner. Det er siden man kommer til via /Registration/... lenker, eller via "Gå til observasjon" på observasjonskort i listevisning.
Detaljsida skal ha en litt annen layout og litt mer info enn i observasjonskortet, men mye er likt.

jorgkv committed:
Første versjon av detaljside

jorgkv committed:
Vis bilder ikke knytta til noen skjema

jorgkv committed:
Litt opprydning

jorgkv committed:
Legg til eget view for snøprofil

jorgkv committed:
Forslag til enklere oppsett

gruble committed:
Lagt inn chips. Skilt ut noen chips i egne komponenter forde de brukes både i kortet og på detaljsida

gruble committed:
Lagt inn alle skjema. Ikke testet hvert skjema ennå

gruble committed:
Alle skjema er lagt inn både for kort og detaljside.

Grafik flyttet øverst i hvert skjema.
Skilt ut endre-knapp i egen komponent for gjenbruk begge steder.

gruble committed:
Laget felles komponent for knapperad.

Fjernet registrert-tidspunkt fra kort.

gruble committed:
Småfiks på bildegrid og justering av linjeskift på tekster

gruble committed:
Fikset bilde-grid slik at den ikke tar plass hvis det ikke er noen bilder i skjema

gruble committed:
Fikset slik at bilder hopper til neste linje hvis det ikke er plass på mobil.

Skilt ut kartkomponenten og gjort det mulig å få opp popup med interaktivt kart også på detaljsida

gruble committed:
Fikset bredde som var altfor smal

gruble committed:
Fikset klikk på bilder som ikke er knyttet til noe skjema

gruble committed:
Fikset slik at skjema for istykkelse vises

gruble committed:
Fikset slik at vi kan vise skjema med bilder selv om man ikke har fylt ut noe i skjemaet.

Måtte hente tittel for skjema på annen måte siden tomme skjema heller ikke har noe summary.

gruble committed:
Småfiks

gruble committed:
Gjort ferdig forslag til feilhåndtering, også med språkstøtte

gruble committed:
Omorganisert feilmeldinger

gruble committed:
Test av ApiInterceptor hører til en annen grein, så sletter den herfra

gruble reviewed (commented):
**NB! Jeg har nettopp rebaset greina på develop.**
Ser jeg har fått med noen endringer som ikke burde vært i greina (som index.html og package-lock), men lart det være. 

Har fått sånn noenlunde orden på feilhåndteringa nå. Å bruke ```registration.hasValue()``` hjalp for å unngå problemer med visning hvis noe feilet.
Jeg synes det er rart at API'et returnerer 204 for GET /Registration, hvis den ikke finner observasjonen, så har laget [en sak på å få endret dette til 404](https://nveprojects.atlassian.net/browse/RO-2997). Da kan vi forenkle koden noe hvis vi ønsker.
Denne PR'en påvirker også observasjonskortet i listevisning, så dette må også testes.

Ting jeg har jobbet med:
- Sjekke at alle skjema er med, for alle naturfarer
- Skille ut kode i komponenter for å kunne gjenbruke dem i både kort- og detaljvisning. F.eks. "knapperad".
- Feilhåndtering
- Endret bittelitt på formatering av lat/lng for å gjøre det lettere å kopiere dette inn i varsom-appen eller f.eks. Google Maps
- Formatert om bildene noe så de ser mer ut som "kort", slik skissene viser. "Portrett-bilder" får fortsatt grå ramme på sidene
- Responsiv formatering av tekst. Summary-komponenten legger nå inn linjeskift mellom hvert felt på smale skjermer. Har beholdt "kulepunkt" mellom hvert felt på bredere skjermer. Jeg synes dette ser bedre ut en å fordele felter i kolonner, slik skissene viser. Noen skjema som har egen visning er ikke i tråd med dette.

Er ikke helt fornøyd med designet, men tror det er ganske tro mot skissene nå, kanskje med unntak av "chipsene":
- Når kartet ikke skal ha sidemarg synes jeg det burde være øverst på sida
- Jeg synes grå bakgrunn i overskrift for hvert skjema gjør at det blir for oppdelt/usammenhengende

Småting jeg vet gjenstår, men som kanskje kan nedprioriteres:
- Vise spinner ved lasting
- Observasjoner med bilder, men uten noe skjema har ikke marg mellom metadata og bilder. Burde også hatt sidemarg. Ellers synes jeg det er helt ok å slenge bilder uten skjematilknytning med på slutten, uten noe overskrift (ser det står en TODO om dette i koden, lar den stå så lenge).

jorgkv committed:
Legg til alt for bilder

jorgkv committed:
Bare lag attachment-grid hvis det finnes bilder

jorgkv committed:
Hent geohazard navn umiddelbart

Legger til en hjelpefunksjon for å hente navn
umiddelbart, uten observable.

Dette forenkler koden litt i geohazard-chip.
Jeg tror ikke det kan skje at oversettelser ikke er lastet når chipen vises.

jorgkv committed:
Kommentar

jorgkv commented on src/app/components/observation/observation/observation.component.css:
```diff
@@ -65,20 +65,15 @@ swiper-container swiper-slide:first-child:nth-last-child(1) {
 
-app-static-map-image {
-  width: 100%;
+observation-location-map {
```
Denne cssen trooooor jeg kan slettes? Komponenten heter vel "app-observation-location-map" så denne regelen er nok ikke i bruk.

jorgkv committed:
Slett kommentar som ikke gjelder lenger

jorgkv commented on src/index.html:
```diff
@@ -12,13 +12,16 @@
 
     <!-- add to homescreen for ios -->
-    <meta name="apple-mobile-web-app-capable" content="yes" />
+    <meta name="mobile-web-app-capable" content="yes" />
```
Det er ikke noe spesiell grunn til at denne er endra? Hvis ikke tror jeg vi bør reversere den, kan ikke se at `mobile-web-app-capable` er gyldig, mens `apple-mobile-web-app-capable` fortsatt har en funksjon på iOS.

```suggestion
    <meta name="apple-mobile-web-app-capable" content="yes" />
```

gruble replied:
Ja, jeg fikk en advarsel i konsollet om at den funksjonen var utdatert. Det var derfor jeg endret den.
<img width="806" height="272" alt="image" src="https://github.com/user-attachments/assets/8bfbfa4e-879e-4ff9-8c49-bb4f48203d96" />

gruble replied:
Men jeg kan fjerne den igjen så kan vi heller ta det seinere evt.

jorgkv reviewed (commented):
Se noen små kommentarer. Jeg kommer også til å sjekke inn noen små commits med små endringer jeg syns var verdt å ta med mens jeg satt med dette her.

Syns mobilvennlig visning i summary.component fungerte bra. Testa på observasjoner med en del data i, syns ikke de blir for lange, det er bedre at det er lettlest. Bra jobba!

Jeg er litt usikker på om hva som bør stå i sidetittelen her:

<img width="594" height="233" alt="image" src="https://github.com/user-attachments/assets/54118ea1-b4fd-49ef-a31f-cdf9b8531c16" />

Det hadde kanskje gitt mer mening om det stod "Observasjon fra ...", men det fungerer ikke på små skjermer kanskje.
Ikke noe vi trenger å fikse nå akkurat dette. Det var jeg som la til det der i tittelen tror jeg, så dette er mer en kommentar til meg selv enn til deg, men kom gjerne med forslag. Koden blir jo enklere om vi bare bytter til "Observasjon", men når man scroller nedover er det egentlig fint å ha sted / dato i tittelen, siden den synes hele tiden, så ikke sikker på hva som er best.

Spinner osv kan vi ta i neste sprint eller når vi har tid.

Jeg kan ikke godkjenne min egen PR, så det må du eventuelt gjøre. Men jeg syns det ser bra ut, godkjent herfra 👍

gruble committed:
Fjernet ubrukt stil og reversert endring av meta-tag 'mobile-web-app-capable'
